docs/user-guide/spark-user-guide.md

```diff
@@ -326,9 +333,13 @@ If the user wishes to limit which jobs are evaluated during a specific applicati
 
-For each aggregation context, a separate output table is created. The storage path for the output table is configured using `-o` options as described above. The records of the context output table consist of the unique key for the specific context record (sharing the same name as the field in the model definition) and a column for each CQL expression that is evaluated. The default column name for the CQL evaluation columns is `<Library Name>|<Expression Name>`. The pipe delimiter can be changed using the `--default-output-column-delimiter` program option or users can provide their own column name alongside the expression in the `cql-jobs.json` file. An output column name for an expression is defined using the format `{"name": "<Expression Name>", "outputColumn": "<Output Column Name>"}` in the `expressions` list rather than using a plain string for an expression. If more than one column ends up with the same name, a runtime error will be thrown.
+For each aggregation context, a separate output table is created. The storage path for the output table is configured using `-o` options as described above. The output table will contain one or more context records per input context ID with a context ID column (sharing the same name as the field in the model definition), a parameters column, and a column for each CQL expression that is evaluated. The default column name for the CQL expression result is `<Library Name>|<Expression Name>`. The pipe delimiter can be changed using the `--default-output-column-delimiter` program option or users can provide their own column name alongside the expression in the `cql-jobs.json` file. An output column name for an expression is defined using the format `{"name": "<Expression Name>", "outputColumn": "<Output Column Name>"}` in the `expressions` list rather than using a plain string for an expression. If more than one column ends up with the same name, a runtime error will be thrown.
+
+The parameters column contains JSON data representing the CQL parameters that were used in the evaluation of the column results. The format of the JSON data is the same as the parameter data in the jobs file. This column allows the results of a single context evaluation to be split into multiple rows of related columns. For example, an input parameter might be `EndDate` for the logic and the desired result is for there to be a separate record in the output for each CQL context + end date.
+
+Row grouping by parameters is enabled by default or can be disabled using the `--disable-row-grouping` flag. When enabled, all parameter values are used as the grouping key. In the case where there are more CQL parameter values than are needed for the grouping key, the `--key-parameters` program argument can be used to limit which parameter values are used as the grouping key. Multiple values can be provided in a comma-separate list or this parameter can be repeated as many times as needed to define the exact set of parameters needed.
```

Comment:
The first sentence should probably be split into two.  Like so:  Row grouping by parameters is enabled by default. It can be disabled by using the `--disable-row-grouping` flag.  I feel this little section could be made more understandable by including  a simple example of what the output looks like when CQL parameters are used.

Reply:
I left the first sentence intact as it read pretty clearly to me and I didn't like how choppy it felt split into two. I did, however, break out the parameters stuff into its own subheading and I added some examples.

Reply:
Looks good.